File: week1-python/Application_Sunglasses_Filter/Load_Images.py
# Import libraries
import cv2
import numpy as np
from dataPath import DATA_PATH

# Load the Face Image
faceImagePath = DATA_PATH + "/images/musk.jpg"
faceImage = cv2.imread(faceImagePath)

#cv2.imshow("Face",faceImage)
cv2.imwrite("../results/face.png",faceImage)

# Load the Sunglass image with Alpha channel
# (http://pluspng.com/sunglass-png-1104.html)
glassimagePath = DATA_PATH + "/images/sunglass.png"
glassPNG = cv2.imread(glassimagePath,-1)

# Resize the image to fit over the eye region
glassPNG = cv2.resize(glassPNG,(300,100))
print("image Dimension ={}".format(glassPNG.shape))

# Separate the Color and alpha channels
glassBGR = glassPNG[:,:,0:3]
glassMask1 = glassPNG[:,:,3]

# Display the images for clarity
#cv2.imshow("Sunglasses Color Channels",glassBGR)
#cv2.imshow("Sunglass Alpha Channel",glassMask1)
#cv2.waitKey(0)
#cv2.destroyAllWindows()
cv2.imwrite("../results/glassBGR.png",glassBGR)
cv2.imwrite("../results/glassMask1.png",glassMask1)

File: week1-cpp/Application_Sunglasses_Filter/Load_Images.cpp
#include <iostream>
#include <opencv2/core.hpp>
#include <opencv2/imgproc.hpp>
#include <opencv2/highgui.hpp>
#include <opencv2/opencv.hpp>
#include "dataPath.hpp"

using namespace std;
using namespace cv;

int main(void){
	// Load the Face Image
	string faceImagePath = DATA_PATH + "/images/musk.jpg";

	Mat faceImage = imread(faceImagePath);
	// Load the Sunglass image with Alpha channel
	// (http://pluspng.com/sunglass-png-1104.html)
	string glassimagePath = DATA_PATH + "/images/sunglass.png";
	
	// Read the image
	Mat glassPNG = imread(glassimagePath,-1);
	// Resize the image to fit over the eye region
	resize(glassPNG,glassPNG, Size(300,100));

	cout << "Image Dimension = " << glassPNG.size() << endl;
	cout << "Number of channels = " << glassPNG.channels();

	// Separate the Color and alpha channels
	Mat glassRGBAChannels[4];
	Mat glassRGBChannels[3];
	split(glassPNG, glassRGBAChannels);

	for (int i = 0; i < 3; i++){
	    // Copy R,G,B channel from RGBA to RGB
	    glassRGBChannels[i] = glassRGBAChannels[i];
	}

	Mat glassBGR, glassMask1;
	// Prepare BGR Image
	merge(glassRGBChannels,3,glassBGR);
	// Alpha channel is the 4th channel in RGBA Image
	glassMask1 = glassRGBAChannels[3];

	imwrite("../results/sunglassRGB.png",glassBGR);
	imwrite("../results/sunglassAlpha.png",glassMask1);

	return 0;
}
